### DISCUSSION TITLE
How to get the structure of a parameter struct ?

### DISCUSSION THREAD

**Discussion by hick** [bpftrace/bpftrace#2133] [Q&A]
I've bought the book "BPF PerformanceTools" by brendan,  and finished my first walking through  of it.  really good book, thx to brendan and related tool-chain contributors .   but as far as I know, brendan recommend the way of reading the source code of a tracee built-in function of a programming language, to get the detail struct of a parameter.  like the bash source code for function  "execute_function" in chapter 12.      good way for trace the bash souce code,  but some programming languages like Python/PHP,  even the built-in functions of them maybe defined in C macro,  or even a macro wrapper  of another macro, not so easy to get the structure of the comiled struct parameter.    

Is there any ideas ,  maybe the bpftrace/BCC way or not,   go get the detail members/basic types of a compiled struct of a function paremeter, without reading the hard-to-reading source code of the interpreter like PHP? ---I just know the var in PHP is a zval struct in C source code of  the PHP interpreter.   some way for reflection?  or just like the PHP function var_dump to dump the structure of a  parameter struct?

**Comment by fbs** [comment]
A debugger like GDB might be your best bet right now. We don't have the DWARF and printing support you need for it. Maybe https://linux.die.net/man/1/pahole can work too
